### PR TITLE
scan-build

### PR DESCRIPTION
**Pull request by spfennell**
Fixes for all of the bugs besides MemoryManager and Swig

### CODE REVIEW

**Comment by spfennell** [comment]
@jmpenn do you approve?

File: trick_source/data_products/Log/MatLab4.cpp [hunk 1]
```diff
@@ -364,6 +364,9 @@ int MatLab4LocateParam( char *file_name , char *param_name , char *time_name ) {
                         case 5:
                                 size = 1 ;
                                 break ;
+                        default :
+                                std::cerr << "Matlab4 " << __FUNCTION__ << "ERROR: invalid matrix size" << std::endl;
+                                size = -1;
```
Comment (pappyvw) [inline trick_source/data_products/Log/MatLab4.cpp]: Do you want a `break` here for consistency?

```suggestion
                                size = -1 ;
                                break ;
```

File: trick_source/data_products/Log/MatLab.cpp [hunk 2]
```diff
@@ -111,7 +111,7 @@ MatLab::MatLab(char * file_name, char * param_name, char * time_name) {
             swap_ = 0;
         } else if (magic == 0x494d) {
             swap_ = 1;
-        }
+        }
```
Comment (pappyvw) [inline trick_source/data_products/Log/MatLab.cpp]: Suggest to undo the added trailing whitespace

```suggestion
        }
```

Comment (spfennell) [reply trick_source/data_products/Log/MatLab.cpp]: fixed, thanks

File: trick_source/data_products/Log/MatLab4.cpp (hunk 1 above)
Comment (spfennell) [reply trick_source/data_products/Log/MatLab4.cpp]: fixed, thanks

**Pull request merged**